tests/run/genexpr_arg_order.py

```diff
@@ -0,0 +1,186 @@
+# mode: run
+# tag: genexpr, py3, py2
+
+from __future__ import print_function
+
+# Tests that function arguments to generator expressions are
+# evaluated in the correct order (even after optimization)
+# WARNING: there may be an amount of luck in this working correctly (since it
+# isn't strictly enforced). Therefore perhaps be prepared to disable these
+# tests if they stop working and aren't easily fixed
+
+import cython
+
+@cython.cfunc
+@cython.returns(cython.int)
+def zero():
+    print("In zero")
+    return 0
+
+@cython.cfunc
+@cython.returns(cython.int)
+def five():
+    print("In five")
+    return 5
+
+@cython.cfunc
+@cython.returns(cython.int)
+def one():
+    print("In one")
+    return 1
+
+def skip_bug(f):
+    if not cython.compiled:
+        return f
+
+@skip_bug  # FIXME - I don't think this is easy to enforce unfortunately, but it is slightly wrong
```

Comment:
I didn't manage to get these working correctly (so I'm skipping them) but equally they don't work correctly on the current master, so it isn't a regression.

They can be a problem for some future person I hope.